pandas/tests/frame/methods/test_describe.py

```diff
@@ -413,3 +413,33 @@ def test_describe_exclude_pa_dtype(self):
             dtype=pd.ArrowDtype(pa.float64()),
         )
         tm.assert_frame_equal(result, expected)
+
+    @pytest.mark.parametrize("percentiles", [None, [], [0.2]])
+    def test_refine_percentiles(self, percentiles):
+        """
+        Test that the percentiles are returned correctly depending on the `percentiles`
+        argument.
+        - The default behavior is to return the 25th, 50th, and 75 percentiles
+        - If `percentiles` is an empty list, no percentiles are returned
+        - If `percentiles` is a non-empty list, only those percentiles are returned
+        """
+        # GH#60550
+        df = DataFrame({"a": np.arange(0, 10, 1)})
+
+        result = df.describe(percentiles=percentiles)
+
+        if percentiles is None:
+            percentiles = [0.25, 0.5, 0.75]
+
+        expected = Series(
+            {
+                "count": len(df.a),
+                "mean": df.a.mean(),
+                "std": df.a.std(),
+                "min": df.a.min(),
+                **{f"{p:.0%}": df.a.quantile(p) for p in percentiles},
+                "max": df.a.max(),
+            },
+        ).to_frame(name="a")
```

Comment:
Can you just create `expected = DataFrame(...)` instead of using `to_frame`?

Reply:
Is the new formulation appropriate? I passed `index=['a']` and had to transpose. Maybe there is a cleaner way such as to avoid the transpose?

Reply:
You can do:

```python
DataFrame(
    [len(df.a), df.a.mean(), ..., **[df.a.quantile(p) for p in percentiles], df.a.max()],
    index=pd.Index(["count", ..., **[f"{p:.0%}" for p in percentiles], ...]),
    column=["a"]
)
```

Reply:
Good point, although not sure it's really cleaner, as the list exhaustion is duplicated and this removes the clarity of the mapping from key to value.

Reply:
I would prefer the suggested method to not exercise other extraneous pandas APIs in this test like transpose or `to_frame`

Reply:
That makes total sense; will update it right now.